pkg/types/gcp/platform.go

```diff
@@ -59,4 +64,20 @@ type Platform struct {
 	// such as the current env OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE
 	// +optional
 	Licenses []string `json:"licenses,omitempty"`
+
+	// PrivateDNSZone Technology Preview.
+	// The PrivateDNSZone should be provided as "projects/{project-id}/managedZones/{managed-zone-name}".
+	// When the PrivateDNSZone is empty, the DNS zones will be created and managed in the Service
+	// Project (ProjectID). When "None", the DNS zones will be externally managed. When PrivateDNSZone
+	// is provided, the DNS zone will be created and managed in the Project where it was created.
+	// +optional
+	PrivateDNSZone string `json:"privateDNSZone,omitempty"`
+
+	// PublicDNSZone Technology Preview.
+	// The PublicDNSZone should be provided as "projects/{project-id}/managedZones/{managed-zone-name}".
```

Comment:
Having users pass the DNS zone in this format does not seem ergonomic to me; and is also inconsistent with other parts of the installconfig, like:
https://github.com/openshift/installer/blob/36e5ac36f6f1b796fdf94a3d4bef384e271d87bb/pkg/types/gcp/machinepools.go#L116-L132

I would suggest we have users pass just the managed-zone name with an optional project-id (which defaults to ic.GCP.ProjectID`). Then you can construct this formatted string when you need it.

This will allow you to get rid of the `ParseZoneName` function.

Reply:
This is a great discussion to have in the enhancement proposal https://github.com/openshift/enhancements/pull/1134.

Reply:
The format used is the `Name` format, as defined by the Google API. I can see where it could be considered technically overloaded by enabling the string to either be the `Name` or the `ID` values. Though, there are several decades of precedence for this; for example most modern operating system commands have accepted filenames given in short, relative, or fully qualified format. As such, I don't particularly see this as an issue. If we choose to break this up into a strict struct, I highly recommend using the `ID` identifier for both the specified managedZone and it's project.

https://cloud.google.com/apis/design/resource_names

Reply:
The Google API for Managed Zone also sets this precedence in that it allows both `Name` and `ID`:
https://cloud.google.com/dns/docs/reference/v1/managedZones/get#parameters